eclipse-theia/theia
# Add DebugAdapterNamedPipeServer and DebugAdapterInlineImplementation

#### What it does
Adds new debug adapter descriptor types to the plugin API
Fixes https://github.com/eclipse-theia/theia/issues/10006

#### How to test
Check out the mock debug extension (https://github.com/Microsoft/vscode-mock-debug) and build it (yarn compile & yarn package) in various debug adapter mode (see extension.ts#21). Also, you need to remove some stuff that we don't support yet from activateMockDebug.ts, here's the changed file:

[activateMockDebug.ts.txt](https://github.com/eclipse-theia/theia/files/7224622/activateMockDebug.ts.txt)

The extension allows to add a launch configuration that will step through any *.md file. You just have to point the config at a concrete file instead of using the "select file" command.

#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
